packages/webapp/src/components/ImagePicker/index.tsx

```diff
@@ -0,0 +1,134 @@
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { ChangeEvent, useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { AddLink } from '../Typography';
+import PureFilePickerWrapper from '../Form/FilePickerWrapper';
+import TextButton from '../Form/Button/TextButton';
+import InputBaseLabel from '../Form/InputBase/InputBaseLabel';
+import { MediaWithAuthentication } from '../../containers/MediaWithAuthentication';
+import { ReactComponent as CameraIcon } from '../../assets/images/farm-profile/camera.svg';
+import { ReactComponent as TrashIcon } from '../../assets/images/farm-profile/trash.svg';
+import { ReactComponent as EditIcon } from '../../assets/images/farm-profile/edit.svg';
+import styles from './styles.module.scss';
+
+export type ImagePickerProps = {
+  onSelectImage: (file: File) => void;
+  onRemoveImage: () => void;
+  label?: string;
+  optional?: boolean;
+  defaultUrl?: string;
+};
+
+export default function ImagePicker({
+  onRemoveImage,
+  onSelectImage,
+  defaultUrl = '',
+  label,
+  optional = true, // false is not yet supported
+}: ImagePickerProps) {
+  const [previewUrl, setPreviewUrl] = useState(defaultUrl);
+  const dropContainerRef = useRef<HTMLDivElement>(null);
+  const { t } = useTranslation();
+
+  const removeImage = () => {
+    onRemoveImage();
+    setPreviewUrl('');
+  };
+
+  const showImage = (file: File) => {
+    const url = URL.createObjectURL(file);
+    setPreviewUrl(url);
+    onSelectImage(file);
+  };
+
+  const handleFileInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (!e.target.files) return;
+    const file = e.target.files[0];
+    showImage(file);
+  };
+
+  const handleDragEvent = (e: React.DragEvent) => {
+    e.preventDefault();
+    if (e.type === 'dragover') return;
+
+    if (e.type === 'dragenter' || e.type === 'dragleave') {
+      dropContainerRef.current?.classList.toggle(styles.dropContainerActive);
+    } else if (e.type === 'drop') {
+      const file = e.dataTransfer?.files[0];
+      if (file) showImage(file);
+    }
+  };
+
+  useEffect(() => {
+    return () => URL.revokeObjectURL(previewUrl);
+  }, [previewUrl]);
+
+  return (
+    <div>
+      {label && <InputBaseLabel label={label} optional={optional} />}
+      {previewUrl ? (
+        <div className={styles.imageContainer}>
+          {previewUrl === defaultUrl ? (
+            <MediaWithAuthentication fileUrls={[defaultUrl]} alt="image thumbnail" />
```

Comment:
This can probably be removed along with `previewUrl === defaultUrl`. The parent component can use the `useMediaWithAuthentication` hook to fetch the private image url and pass it as `defaultUrl` to `ImagePicker`.